src/app/domain/osReceive/models/OsReceiveState.ts

```diff
@@ -0,0 +1,75 @@
+import { AcceptFromFlagshipManifest } from './OsReceiveCozyApp'
+
+import { ReceivedFile } from '/app/domain/osReceive/models/ReceivedFile'
+
+export enum OsReceiveActionType {
+  SetFilesToUpload = 'SET_FILES_TO_UPLOAD',
+  SetRouteToUpload = 'SET_ROUTE_TO_UPLOAD',
+  SetFlowErrored = 'SET_FLOW_ERRORED',
+  SetRecoveryState = 'SET_RECOVERY_STATE',
+  SetInitialState = 'SET_INITIAL_STATE',
+  UpdateFileStatus = 'UPDATE_FILE_STATUS',
+  SetCandidateApps = 'SET_CANDIDATE_APPS'
+}
+
+export enum OsReceiveFileStatus {
+  toUpload,
+  uploading,
+  uploaded,
+  error,
+  queued
+}
+
+export interface OsReceiveFile {
+  name: string
+  file: ReceivedFile
+  status: OsReceiveFileStatus
+  handledTimestamp?: number // Unix timestamp representing when the file was handled
+  source?: string // base64 of the file content
+  type?: string // mimetype of the file
+}
+
+export interface OsReceiveState {
+  filesToUpload: OsReceiveFile[]
+  routeToUpload: { href?: string; slug?: string }
+  errored: boolean
+  candidateApps?: AcceptFromFlagshipManifest[]
+}
+
+export type OsReceiveAction =
+  | { type: OsReceiveActionType.SetFilesToUpload; payload: OsReceiveFile[] }
+  | {
+      type: OsReceiveActionType.SetRouteToUpload
+      payload: { href: string; slug: string }
+    }
+  | { type: OsReceiveActionType.SetFlowErrored; payload: boolean }
+  | { type: OsReceiveActionType.SetRecoveryState }
+  | { type: OsReceiveActionType.SetInitialState }
+  | {
+      type: OsReceiveActionType.UpdateFileStatus
+      payload: Omit<OsReceiveFile, 'file'>
+    }
+  | {
+      type: OsReceiveActionType.SetCandidateApps
+      payload: AcceptFromFlagshipManifest[]
+    }
+
+export interface ServiceResponse<T> {
+  result?: T
+  error?: string
+}
+
+export interface OsReceiveApiMethods {
+  getFilesToHandle: (
+    base64: boolean,
+    state: OsReceiveState
+  ) => Promise<OsReceiveFile[]>
+  hasFilesToHandle: () => Promise<UploadStatus>
+  uploadFiles: (arg: string) => Promise<void>
+  resetFilesToHandle: () => Promise<boolean>
+  cancelUploadByCozyApp: () => boolean
```

Comment:
I just noticed that the `boolean` result is never used on cozy-apps. Why do we need it on those methods?

Reply:
> I just noticed that the `boolean` result is never used on cozy-apps. Why do we need it on those methods?

Technically yes it's not used but it is still sent by post-me. I figured it's always better to return true instead of undefined if a post-me method succeeded but has no data to send back